src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -400,46 +432,50 @@ def _check_value(self, action, value):  # pylint: disable=too-many-statements, t
-                                     "If the command is from an extension, "
-                                     "please make sure the corresponding extension is installed. "
-                                     "To learn more about extensions, please visit "
-                                     "{extensions_link}").format(msg=error_msg, extensions_link=extensions_link)
+                    error_msg = "'{value}' is misspelled or not recognized by the system.".format(value=value)
+                az_error = AzCLIError(AzCLIErrorType.CommandNotFoundError, error_msg, command=self.prog)
+
             else:
                 # `command_source` indicates command values have been parsed, value is an argument
                 parameter = action.option_strings[0] if action.option_strings else action.dest
-                error_msg = "{prog}: '{value}' is not a valid value for '{param}'. See '{prog} --help'.".format(
+                error_msg = "{prog}: '{value}' is not a valid value for '{param}'.".format(
                     prog=self.prog, value=value, param=parameter)
                 candidates = difflib.get_close_matches(value, action.choices, cutoff=0.7)
+                az_error = AzCLIError(AzCLIErrorType.ArgumentParseError, error_msg, command=self.prog)
 
-            telemetry.set_user_fault(error_msg)
-            with CommandLoggerContext(logger):
-                logger.error(error_msg)
-            if not caused_by_extension_not_installed:
-                if candidates:
-                    print_args = {
-                        's': 's' if len(candidates) > 1 else '',
-                        'verb': 'are' if len(candidates) > 1 else 'is',
-                        'value': value
-                    }
-                    self._suggestion_msg.append("\nThe most similar choice{s} to '{value}' {verb}:"
-                                                .format(**print_args))
-                    self._suggestion_msg.append('\n'.join(['\t' + candidate for candidate in candidates]))
+            _, params, extension = self._get_failure_recovery_arguments()
+            if candidates:
+                az_error.set_recommendation("Did you mean '{}' ?".format(candidates[0]))
+
+            # recommand a command for user
+            recommender = CommandRecommender(command_name_inferred[3:].strip(), params, extension, error_msg, cli_ctx)
+            recommender.set_help_examples(self.get_examples(command_name_inferred))
+            recommended_command = recommender.recommend_a_command()
+            if recommended_command:
+                az_error.set_recommendation("Try this: '{}'".format(recommended_command))
 
+            # remind user to check extensions if we can not find a command to recommend
+            if az_error.error_type == AzCLIErrorType.CommandNotFoundError \
+                    and not az_error.recommendations and self.prog == 'az':
+                az_error.set_recommendation(EXTENSION_REFERENCE)
```

Comment:
EXTENSION_REFERENCE should not be shown if dynamic extension install is turned on and found no match for the command in the extension command tree.

Reply:
Add `use_dynamic_install == 'no'` in the condition.

Reply:
Modified.